lib/src/internal/FlowManager.cpp

```diff
@@ -305,8 +451,28 @@ namespace mxl::lib
         auto uuid = uuids::to_string(flowId);
         MXL_TRACE("Delete flow: {}", uuid);
 
-        auto const base = std::filesystem::path{_mxlDomain};
-        return (remove_all(makeFlowDirectoryName(base, uuid)) != 0);
+        // Compute the flow directory path
+        auto const flowPath = makeFlowDirectoryName(_mxlDomain, uuid);
+        try
+        {
+            auto const removed = std::filesystem::remove_all(flowPath);
+            if (removed == 0)
+            {
+                MXL_WARN("Flow not found or already deleted: {}", uuid);
+                return false;
+            }
+            return true;
+        }
+        catch (std::filesystem::filesystem_error const& e)
```

Comment:
Please catch `std::exception` instead, because the call may also show `std::bad_alloc`.